pkg/policies/scorecard/scorecard.go

```diff
@@ -0,0 +1,308 @@
+
+	// Threshold is the score threshold that checks must meet to pass the
+	// policy. If all checks score equal or above the threshold, the Allstar
+	// policy will pass. The default is checker.MaxResultScore:
+	// https://pkg.go.dev/github.com/ossf/scorecard/v4@v4.4.0/checker#pkg-constants
+	Threshold int `json:"threshold"`
+}
+
+// RepoConfig is the repo-level config for this policy.
+type RepoConfig struct {
+	// OptConfig is the standard repo-level opt in/out config.
+	OptConfig config.RepoOptConfig `json:"optConfig"`
+
+	// Action overrides the same setting in org-level, only if present.
+	Action *string `json:"action"`
+
+	// Checks overrides the same setting in org-level, only if present.
+	Checks *[]string `json:"checks"`
+
+	// Threshold overrides the same setting in org-level, only if present.
+	Threshold *int `json:"threshold"`
+}
+
+type mergedConfig struct {
+	Action    string
+	Checks    []string
+	Threshold int
+}
+
+type details struct {
+	// Findings key is the check name, and value are logs from Scorecards.
+	Findings map[string][]string
+}
+
+var configFetchConfig func(context.Context, *github.Client, string, string, string, config.ConfigLevel, interface{}) error
+var doNothingOnOptOut = operator.DoNothingOnOptOut
+
+func init() {
+	configFetchConfig = config.FetchConfig
+}
+
+// Scorecard is the Security Scorecard policy object, implements
+// policydef.Policy.
+type Scorecard bool
+
+// NewScorecard returns a new Scorecard policy.
+func NewScorecard() policydef.Policy {
+	var b Scorecard
+	return b
+}
+
+// Name returns the name of this policy, implementing policydef.Policy.Name()
+func (b Scorecard) Name() string {
+	return polName
+}
+
+// Check performs the policy check for this policy based on the
+// configuration stored in the org/repo, implementing policydef.Policy.Check()
+func (b Scorecard) Check(ctx context.Context, c *github.Client, owner,
+	repo string) (*policydef.Result, error) {
+	oc, orc, rc := getConfig(ctx, c, owner, repo)
+	enabled, err := config.IsEnabled(ctx, oc.OptConfig, orc.OptConfig, rc.OptConfig, c, owner, repo)
+	if err != nil {
+		return nil, err
+	}
+	log.Info().
+		Str("org", owner).
+		Str("repo", repo).
+		Str("area", polName).
+		Bool("enabled", enabled).
+		Msg("Check repo enabled")
+	if !enabled && doNothingOnOptOut {
+		// Don't run this policy if disabled and requested by operator. This is
+		// only checking enablement of policy, but not Allstar overall, this is
+		// ok for now.
+		return &policydef.Result{
+			Enabled:    enabled,
+			Pass:       true,
+			NotifyText: "Disabled",
+			Details:    details{},
+		}, nil
+	}
+	mc := mergeConfig(oc, orc, rc, repo)
+
+	fullName := fmt.Sprintf("%s/%s", owner, repo)
+	tr := c.Client().Transport
+	scc, err := scorecard.Get(ctx, fullName, tr)
+	if err != nil {
+		return nil, err
+	}
+
+	var notify string
+	pass := true
+	f := make(map[string][]string)
+
+	for _, n := range mc.Checks {
+
+		l := checker.NewLogger()
```

Comment:
Is it possible to move everything from here to convertLogs into its own function so that you can test more thoroughly?

Reply:
I ended up leaving this, but did add a test for Check.